.github/workflows/npm-publish.yaml

```diff
@@ -0,0 +1,38 @@
+# This workflow will run tests using node and then publish a package to GitHub Packages when a release is created
+# For more information see: https://docs.github.com/en/actions/publishing-packages/publishing-nodejs-packages
+
+name: Node.js Package
+
+on:
+  push:
+    tags:
+      - v**
+
+
+jobs:
+  build:
```

Comment:
Can be done with a single job

Reply:
Get rid of `build` job and add `pnpm test` step to the publish job?

Reply:
Yep!